hs_core/views/__init__.py

```diff
@@ -208,6 +211,15 @@ def update_metadata_element(request, shortkey, element_name, element_id, *args,
 
     return HttpResponseRedirect(request.META['HTTP_REFERER'])
 
+@api_view(['GET'])
+def file_download_url_mapper(request, shortkey, filename):
+    """ maps the file URIs in resourcemap document to django_irods download view function"""
+
+    authorize(request, shortkey, view=True, edit=True, full=True, superuser=True)
```

Comment:
@pkdash your comments remind me that authorize() function itself raises exception if not authorized by default although the calling function can disable this exception-raising behavior. So you are right, in this case, this authorize() call is needed and it will not get to response redirect at all if not authorized since an exception will be raised from authorize() function. Now I got it and agree it is not an issue.

I also got your point on using the api_view decorator here, and I assume even without the api_view decorator in django_irods download() view, this url mapper will be able to pass in the correct request with the right user to download() view function, right? If this is indeed true, my question is for @selimnairb then: is it possible you can use the similar way for your api call as @pkdash did here? i.e., remove api_view decorator from download() view function in django_irods, but put this decorator in the api view as @pkdash did here? 

In any case, I give +1 here for this branch to be merged into develop.